app/controllers/api/v1/testimonials_controller.rb

```diff
@@ -11,45 +11,49 @@ def index
 
   def create
     @testimonial = Testimonial.new(testimonial_params)
-    if admin?(@current_user) && @testimonial.save
-      render json: TestimonialSerializer.new(@testimonial).serializable_hash.to_json, status: :created
+    if admin?(@current_user) 
+      if @testimonial.save
+        render json: TestimonialSerializer.new(@testimonial).serializable_hash.to_json, status: :created
+      else
+        render json: @testimonial.errors, status: :unprocessable_entity
+      end
     else
-      render json: @testimonial.errors, status: :unprocessable_entity
+      render json: {msg: 'You are not authorized to perform that action'}, status: :unauthorized
     end
   end
   
   def update
-    if admin?(@current_user) && @testimonial.update(testimonial_params)
-      render json: TestimonialSerializer.new(@testimonial).serializable_hash.to_json, status: :created
+    if admin?(@current_user) 
+      if @testimonial.update(testimonial_params)
+        render json: TestimonialSerializer.new(@testimonial).serializable_hash.to_json, status: :created
+      else
+        render json: @testimonial.errors, status: :unprocessable_entity
+      end
     else
-      render json: @testimonial.errors, status: :unprocessable_entity
+      render json: {msg: 'You are not authorized to perform that action'}, status: :unauthorized
```

Comment:
Idem anterior con el espaciado de las llaves.
```suggestion
      render json: { msg: 'You are not authorized to perform that action' }, status: :unauthorized
```